File: isPalindromeUsingRecursion/isPalindromeUsingRecursion.js
const isPalindrome=(s,i=0)=>{
    let n=s.length;

    if(i>=Math.floor(n/2)){
        return true;
    }
    
    if(s[i]!=s[n-1-i]){
        return false;
    }
    return isPalindrome(s,i+1);
}
let s="madam";
let ans=isPalindrome(s);
console.log(ans);

File: isPalindromeUsingRecursion/isPalindromeusingRecursion.py
def isPalindrome(s,i):
    n=len(s)
    if i>=n//2:
        return True
    if s[i]!=s[n-1-i]:
        return False
    return isPalindrome(s,i+1)

s='madam'
print(isPalindrome(s,0))
